src/Compilers/Core/Portable/WellKnownMember.cs

```diff
@@ -498,6 +498,7 @@ internal enum WellKnownMember
         System_ReadOnlySpan_T__get_Item,
         System_ReadOnlySpan_T__get_Length,
         System_ReadOnlySpan_T__Slice_Int_Int,
+        System_ReadOnlySpan_T__GetPinnableReference,
```

Comment:
>System_ReadOnlySpan_T__GetPinnableReference [](http://example.com/codeflow?start=8&length=43)

Is there a specific reason to use this helper? It looks like we can use System_ReadOnlySpan_T__get_Item to get a reference to the first item

Reply:
> I'm not sure which of these is better in practice.

It sounds like we need to make a decision and the decision should have a solid rational behind it.

Reply:
> ```get_Item``` needs to perform a bounds check

This is an issue only if the span is empty. But if it is empty, do we even want to go through all the trouble only to copy nothing at the end?

Reply:
> But if it is empty, do we even want to go through all the trouble only to copy nothing at the end?

We don't perform the optimization if the span is empty. But the bounds check is still executed even if the span is not empty (and the check succeeds), so it's wasted time (unless JIT optimizes this).

> It sounds like we need to make a decision and the decision should have a solid rational behind it.

I can do a benchmark if we want to make that decision based on performance characteristics. Otherwise it seems the choices here are interchangeable from compiler's perspective.

Reply:
> Otherwise it seems the choices here are interchangeable from compiler's perspective.

If we don't have strong reasons to choose one over the other. My recommendation is to stick with helper we already use. Especially that the tests show, that we are not going to use it for empty spans.